cmd/notation/blob/inspect.go

```diff
@@ -0,0 +1,156 @@
+		Long:  longMessage,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing signature path to the artifact: use `notation blob inspect --help` to see what parameters are required")
+			}
+			opts.signaturePath = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBlobInspect(opts)
+		},
+	}
+
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	cmd.SetPflagOutput(command.Flags(), &opts.outputFormat, cmd.PflagOutputUsage)
+	return command
+}
+
+func runBlobInspect(opts *blobInspectOpts) error {
+	if opts.outputFormat != cmd.OutputJSON && opts.outputFormat != cmd.OutputPlaintext {
+		return fmt.Errorf("unrecognized output format %s", opts.outputFormat)
+	}
+
+	// initialize
+	mediaType, err := envelope.GetEnvelopeMediaType(filepath.Ext(opts.signaturePath))
+	if err != nil {
+		return err
+	}
+	contents, err := readFile(opts.signaturePath)
+	if err != nil {
+		return err
+	}
+	output := outputs.InspectOutput{MediaType: mediaType, Signatures: []outputs.SignatureOutput{}}
+	skippedSignatures := false
+	err, skippedSignatures, output.Signatures = outputs.Signature(mediaType, skippedSignatures, "nil", output, contents)
+	if err != nil {
+		return nil
+	}
+	if err := printOutput(opts.outputFormat, opts.signaturePath, output); err != nil {
+		return err
+	}
+	return nil
+}
+
+func printOutput(outputFormat string, ref string, output outputs.InspectOutput) error {
+	if outputFormat == cmd.OutputJSON {
+		return ioutil.PrintObjectAsJSON(output)
+	}
+
+	if len(output.Signatures) == 0 {
+		fmt.Printf("%s has no associated signature\n", ref)
+		return nil
+	}
+
+	root := tree.New(ref)
+	var signature outputs.SignatureOutput
+	root.AddPair("signature algorithm", signature.SignatureAlgorithm)
+	root.AddPair("signature envelope type", signature.MediaType)
+
+	signedAttributesNode := root.Add("signed attributes")
+	outputs.AddMapToTree(signedAttributesNode, signature.SignedAttributes)
+
+	unsignedAttributesNode := root.Add("unsigned attributes")
+	outputs.AddMapToTree(unsignedAttributesNode, signature.UnsignedAttributes)
+
+	certListNode := root.Add("certificates")
+	for _, cert := range signature.Certificates {
+		certNode := certListNode.AddPair("SHA256 fingerprint", cert.SHA256Fingerprint)
+		certNode.AddPair("issued to", cert.IssuedTo)
+		certNode.AddPair("issued by", cert.IssuedBy)
+		certNode.AddPair("expiry", cert.Expiry)
+	}
+
+	artifactNode := root.Add("signed artifact")
+	artifactNode.AddPair("media type", signature.SignedArtifact.MediaType)
+	artifactNode.AddPair("digest", signature.SignedArtifact.Digest.String())
+	artifactNode.AddPair("size", strconv.FormatInt(signature.SignedArtifact.Size, 10))
+
+	root.Print()
+	return nil
+}
+
+func readFile(path string) ([]byte, error) {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	size, err := os.Stat(path)
+	if size.Size() == 0 {
+		return nil, fmt.Errorf("file is empty")
+	}
```

Comment:
Why do we care file size, can we simplify code?

Reply:
File size will be useful to check if file is not empty. 
Should we not do this check?

Reply:
Given that you have already read file, Is there an more readable/maintainable way?

Reply:
done